src/Microsoft.ML.TimeSeries/STL/Loess.cs

```diff
@@ -0,0 +1,204 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Text;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.TimeSeries
+{
+    /// <summary>
+    /// Loess is short for Robust Locally Weighted Regression and Smoothing Scatterplots.
+    /// </summary>
+    public class Loess
+    {
+
+        private const double NumericalThreshold = 1.0e-10;
+
+        /// <summary>
+        /// the ratio to determine the local region
+        /// </summary>
+        private readonly int _r;
+
+        private readonly bool _isTemporal;
+
+        /// <summary>
+        /// key is the index of the given point, value is the corresponding neighbors of the given point.
+        /// </summary>
+        private readonly Dictionary<int, LocalRegression> _neighbors = new Dictionary<int, LocalRegression>();
```

Comment:
The CI is failing, and among the errors it's that you can't initialize this member outside the constructor
`Member _neighbors has a field initializer outside the constructor`

Reply:
Updated.

---
In reply to: [435637442](https://github.com/dotnet/machinelearning/pull/5202#discussion_r435637442) [](ancestors = 435637442)